From 71207ec4de6ee7fabe06c21f02bbf0785bc308a9 Mon Sep 17 00:00:00 2001
From: Lionel Le Folgoc <mrpouit@ubuntu.com>
Date: Wed, 20 Jan 2010 20:10:29 +0100
Subject: [PATCH] Assume Hal is always available (it'll be spawned by dbus on demand anyway).

---
 src/xfpm-dbus-monitor.c |    3 +--
 1 files changed, 1 insertions(+), 2 deletions(-)

diff --git a/src/xfpm-dbus-monitor.c b/src/xfpm-dbus-monitor.c
index b77b21d..95b66f7 100644
--- a/src/xfpm-dbus-monitor.c
+++ b/src/xfpm-dbus-monitor.c
@@ -349,8 +349,7 @@ xfpm_dbus_monitor_init (XfpmDBusMonitor *monitor)
 				monitor, 
 				NULL);
 				
-    monitor->priv->hal_connected  = xfpm_dbus_name_has_owner (dbus_g_connection_get_connection (monitor->priv->system_bus), 
-							      "org.freedesktop.Hal");
+    monitor->priv->hal_connected  = TRUE;
 							      
     xfpm_dbus_monitor_add_service (monitor, DBUS_BUS_SESSION, "org.freedesktop.Hal");
 }
-- 
1.6.3.3

