diff --git a/src/xfpm-manager.c b/src/xfpm-manager.c
index a7417a5..96d042c 100644
--- a/src/xfpm-manager.c
+++ b/src/xfpm-manager.c
@@ -513,13 +513,13 @@ xfpm_manager_get_systemd_events(XfpmManager *manager)
         LOGIND_HANDLE_LID_SWITCH, &handle_lid_switch,
         NULL);
 
-    if (handle_power_key)
+    if (!handle_power_key)
         events = g_slist_append(events, "handle-power-key");
-    if (handle_suspend_key)
+    if (!handle_suspend_key)
         events = g_slist_append(events, "handle-suspend-key");
-    if (handle_hibernate_key)
+    if (!handle_hibernate_key)
         events = g_slist_append(events, "handle-hibernate-key");
-    if (handle_lid_switch)
+    if (!handle_lid_switch)
         events = g_slist_append(events, "handle-lid-switch");
 
     while (events != NULL)
