diff --git a/src/xfpm-manager.c b/src/xfpm-manager.c
index a7417a5..96d042c 100644
--- a/src/xfpm-manager.c
+++ b/src/xfpm-manager.c
@@ -513,13 +513,13 @@ xfpm_manager_get_systemd_events(XfpmManager *manager)
         LOGIND_HANDLE_LID_SWITCH, &handle_lid_switch,
         NULL);
 
-    if (handle_power_key)
+    if (!handle_power_key)
         events = g_slist_append(events, "handle-power-key");
-    if (handle_suspend_key)
+    if (!handle_suspend_key)
         events = g_slist_append(events, "handle-suspend-key");
-    if (handle_hibernate_key)
+    if (!handle_hibernate_key)
         events = g_slist_append(events, "handle-hibernate-key");
-    if (handle_lid_switch)
+    if (!handle_lid_switch)
         events = g_slist_append(events, "handle-lid-switch");
 
     while (events != NULL)
diff --git a/src/xfpm-xfconf.c b/src/xfpm-xfconf.c
index 2603bc1..721d977 100644
--- a/src/xfpm-xfconf.c
+++ b/src/xfpm-xfconf.c
@@ -563,7 +563,7 @@ xfpm_xfconf_class_init (XfpmXfconfClass *klass)
                                      PROP_LOGIND_HANDLE_POWER_KEY,
                                      g_param_spec_boolean (LOGIND_HANDLE_POWER_KEY,
                                                            NULL, NULL,
-                                                           TRUE,
+                                                           FALSE,
                                                            G_PARAM_READWRITE));
 
     /**
@@ -573,7 +573,7 @@ xfpm_xfconf_class_init (XfpmXfconfClass *klass)
                                      PROP_LOGIND_HANDLE_SUSPEND_KEY,
                                      g_param_spec_boolean (LOGIND_HANDLE_SUSPEND_KEY,
                                                            NULL, NULL,
-                                                           TRUE,
+                                                           FALSE,
                                                            G_PARAM_READWRITE));
 
     /**
@@ -583,7 +583,7 @@ xfpm_xfconf_class_init (XfpmXfconfClass *klass)
                                      PROP_LOGIND_HANDLE_HIBERNATE_KEY,
                                      g_param_spec_boolean (LOGIND_HANDLE_HIBERNATE_KEY,
                                                            NULL, NULL,
-                                                           TRUE,
+                                                           FALSE,
                                                            G_PARAM_READWRITE));
 
     /**
@@ -593,7 +593,7 @@ xfpm_xfconf_class_init (XfpmXfconfClass *klass)
                                      PROP_LOGIND_HANDLE_LID_SWITCH,
                                      g_param_spec_boolean (LOGIND_HANDLE_LID_SWITCH,
                                                            NULL, NULL,
-                                                           TRUE,
+                                                           FALSE,
                                                            G_PARAM_READWRITE));
 
     g_type_class_add_private (klass, sizeof (XfpmXfconfPrivate));
