Description: Fix linking with --no-add-needed option
Origin: backport, http://git.xfce.org/apps/xfce4-power-manager/commit/?id=7f977d69a0ca4c17dc27554762f50acee1530b07
Bug: http://bugzilla.xfce.org/show_bug.cgi?id=6954
--- xfce4-power-manager-0.8.5.orig/libxfpm/Makefile.am
+++ xfce4-power-manager-0.8.5/libxfpm/Makefile.am
@@ -47,6 +47,11 @@ libxfpmcommon_la_CFLAGS =       	\
 	$(LIBXFCEGUI4_CFLAGS)		\
 	$(LIBNOTIFY_CFLAGS)
 
+libxfpmcommon_la_LIBADD =		\
+	$(GTK_LIBS)					\
+	$(GLIB_LIBS)				\
+	$(DBUS_LIBS)
+
 if MAINTAINER_MODE
 
 BUILT_SOURCES =				\
