diff --git a/src/xfpm-notify.c b/src/xfpm-notify.c
index c93a843..3c2651d 100644
--- a/src/xfpm-notify.c
+++ b/src/xfpm-notify.c
@@ -233,11 +233,11 @@ xfpm_notify_new_notification_internal (const gchar *title, const gchar *message,
 #if !NOTIFY_CHECK_VERSION (0, 7, 0) 
     if ( icon )
     	notify_notification_attach_to_status_icon (n, icon);
+#endif
 #else
     if ( icon )
     	notify_notification_attach_to_status_icon (n, icon);
 #endif
-#endif
 	
     notify_notification_set_urgency (n, (NotifyUrgency)urgency);
     
