index 763b014..6b1bd76 100644
--- a/src/xfpm-battery.c
+++ b/src/xfpm-battery.c
@@ -320,15 +320,15 @@ xfpm_battery_refresh_icon (XfpmBattery *
     {
 	if ( !battery->priv->present || battery->priv->state == XFPM_DEVICE_STATE_EMPTY )
 	{
-	    g_snprintf (icon_name, 128, "%s-000", battery->priv->icon_prefix);
+	    g_snprintf (icon_name, 128, "%s000", battery->priv->icon_prefix);
 	}
 	else if ( battery->priv->state == XFPM_DEVICE_STATE_FULLY_CHARGED )
 	{
-	    g_snprintf (icon_name, 128, "%s-100", battery->priv->icon_prefix);
+	    g_snprintf (icon_name, 128, "%s100", battery->priv->icon_prefix);
 	}
 	else if ( battery->priv->state == XFPM_DEVICE_STATE_DISCHARGING )
 	{
-	    g_snprintf (icon_name, 128, "%s-%s", 
+	    g_snprintf (icon_name, 128, "%s%s", 
 			battery->priv->icon_prefix, 
 			xfpm_battery_get_icon_index (battery->priv->type, battery->priv->percentage));
 	}
